unittests/lit.site.cfg.py.in

```diff
@@ -0,0 +1,20 @@
+@LIT_SITE_CFG_IN_HEADER@
+
+import sys
+
+config.llvm_src_root = "@LLVM_SOURCE_DIR@"
+config.llvm_obj_root = "@LLVM_BINARY_DIR@"
+config.llvm_build_mode = "@LLVM_BUILD_MODE@"
+config.pstore_obj_root = "@PSTORE_BINARY_DIR@"
```

Comment:
Should there be a definition using PSTORE_SOURCE_DIR as well (largely for symmetry)?

Reply:
I don't hold with symmetry being a good reason to add unused code myself.

Reply:
Ok, I won't press it, but I would point out a couple of things:
- In my experience symmetrical code is also correct code. Code is easier to maintain if symmetry is exposed where it exists, and where asymmetry is emphasised when it is present.
- There is a very, very, fine line between adding an unused assignment and adding a redundant variable definition (`PSTORE_SOURCE_DIR`/`PSTORE_ROOT_DIR` above).

Reply:
It looks like that the `LLVM/Clang` projects don’t use `LLVM_ROOT_DIR` or `CLANG_ROOT_DIR`. I checked the CMake document, I could find the definition of `PROJECT_SOURCE_DIR` (top level source directory for the current project), but I can’t find the `PROJECT_ROOT_DIR`. Should we only use `PROJECT_SOURCE_DIR` for the pstore project?

Reply:
> In my experience symmetrical code is also correct code. Code is easier to maintain if symmetry is exposed where it exists, and where asymmetry is emphasised when it is present.

I largely agree, you'd expect things to be balanced. In this case tests can use things from the build directory and sometimes need things from the source side. In this case the latter is not required.

> Would it have made any difference if I hard written: “Should there be a definition using PSTORE_SOURCE_DIR as well for consistency with the LLVM variables?”

I read that into your your original comment, and it did make me have a look through and consider whether it should be there. So it was a good point to raise in review.